bytesconv.go

```diff
@@ -393,6 +393,66 @@ func s2b(s string) (b []byte) {
 	return b
 }
 
+var quotedArgShouldEscapeTable = func() [256]bool {
+	// According to RFC 3986 §2.3
+	var a [256]bool
+	for i := 0; i < 256; i++ {
+		a[i] = true
+	}
+
+	// ALPHA
+	for i := int('a'); i <= int('z'); i++ {
+		a[i] = false
+	}
+	for i := int('A'); i <= int('Z'); i++ {
+		a[i] = false
+	}
+
+	// DIGIT
+	for i := int('0'); i <= int('9'); i++ {
+		a[i] = false
+	}
+
+	// Unreserved characters
+	a[int('-')] = false
+	a[int('_')] = false
+	a[int('.')] = false
+	a[int('~')] = false
+
+	return a
+}()
+
+var quotedPathShouldEscapeTable = func() [256]bool {
+	// From the spec: http://tools.ietf.org/html/rfc3986#section-3.3
+	// an path can contain zero or more of pchar that is defined as follows:
+	// pchar       = unreserved / pct-encoded / sub-delims / ":" / "@"
+	// pct-encoded = "%" HEXDIG HEXDIG
+	// unreserved  = ALPHA / DIGIT / "-" / "." / "_" / "~"
+	// sub-delims  = "!" / "$" / "&" / "'" / "(" / ")"
+	//             / "*" / "+" / "," / ";" / "="
```

Comment:
Is this comment still relevant? I see characters such as `!` which aren't present in your new code?